src/components/Buttons/Button.js

```diff
@@ -0,0 +1,266 @@
+    fontSize: '14px',
+    height: '32px'
+  },
+
+  standard: {
+    ...spacing.PADDING_X_SM,
+    fontSize: '16px',
+    height: '40px'
+  },
+
+  large: {
+    ...spacing.PADDING_X_MD,
+    fontSize: '18px',
+    height: '48px'
+  }
+}
+
+const linkStyles = {
+  display: 'flex',
+  alignItems: 'center',
+  boxSizing: 'border-box'
+}
+
+const getSnacksStyles = props => {
+  const { action, actionHover, primaryBackground } = props.snacksTheme.colors
+
+  return {
+    primary: {
+      base: {
+        backgroundColor: action,
+        color: primaryBackground,
+
+        ':hover': {
+          backgroundColor: actionHover
+        },
+        ':active': {
+          backgroundColor: actionHover
+        }
+      },
+
+      inverted: {
+        backgroundColor: primaryBackground,
+        border: `1px solid ${action}`,
+        color: action,
+
+        ':hover': {
+          opacity: '0.9'
+        },
+        ':active': {
+          opacity: '0.8'
+        }
+      }
+    },
+
+    secondary: {
+      base: {
+        backgroundColor: 'transparent',
+        color: action,
+        border: `1px solid ${action}`,
+
+        ':hover': {
+          color: actionHover,
+          border: `1px solid ${actionHover}`
+        },
+        ':active': {
+          color: actionHover,
+          border: `1px solid ${actionHover}`
+        }
+      },
+
+      inverted: {
+        color: primaryBackground,
+        border: `1px solid ${primaryBackground}`,
+
+        ':hover': {
+          opacity: '0.9'
+        },
+        ':active': {
+          opacity: '0.8'
+        }
+      }
+    },
+
+    coupon: {
+      base: {
+        backgroundColor: colors.WHITE,
+        color: colors.RED_500,
+        border: `1px dashed ${colors.GRAY_88}`,
+
+        ':hover': {
+          border: `1px dashed ${colors.RED_600}`,
+          color: colors.RED_600
+        },
+        ':active': {
+          border: `1px dashed ${colors.RED_700}`,
+          color: colors.RED_700
+        }
+      }
+    },
+
+    disabled: {
+      base: {
+        backgroundColor: colors.GRAY_74,
+        color: colors.WHITE,
+        cursor: 'not-allowed'
+      }
+    }
+  }
+}
+
+const Button = props => {
+  // Disabled styles override all other snacks-specific button styles
+  const snacksStyle = props.disabled ? 'disabled' : props.snacksStyle
+  const snacksStyles = getSnacksStyles(props)
+
+  const ElementType = props.href ? 'a' : 'button'
+
+  const finalProps = {
+    disabled: props.disabled,
+    tabIndex: props.tabIndex,
+    type: props.type,
+    style: [
+      baseStyles,
+      sizeStyles[props.size],
+      snacksStyles[snacksStyle].base,
+      props.inverted && snacksStyles[snacksStyle].inverted,
+      ElementType === 'a' && linkStyles,
+      props.styles
+    ],
+    onClick: e => {
+      e.preventDefault()
+      props.onClick()
+    },
+    ...props.extraProps
+  }
+  if (props.href) {
+    finalProps.href = props.href
+  }
+
+  let icon = props.icon
```

Comment:
Yesterday, dom was talking about how we want to prefer const to let:

```
const icon = typeof props.icon === 'string' ? <Icon name={props.icon} /> : props.icon
```

Reply:
I went back and forth between the two, I felt like this was more readable but I'm like 51/49, so I'll go with whatever the group wants @Eldwick @dcocchia

Reply:
We should only prefer it when its unnecessary to use `let` or `var`. I think using `let` to increase readability, or for actual reassignments, is ok. Of course that's a bit subjective :D